packages/badge-service/badgeGenerator/badgeGenerator.ts

```diff
@@ -0,0 +1,74 @@
+const sha512 = require('js-sha512');
+import { MutationScoreMapper, MutationScore } from 'stryker-dashboard-data-access';
+import * as httpHelpers from '../helpers/httpHelpers';
+
+export async function run(context: any, req: any) {
+    let statusCode = 400;
+    context.res = {
+        status: statusCode,
+    };
+
+    const scoreRepo = new MutationScoreMapper();
+
+    let rowKey = context.bindingData.repo;
+
+    if (context.bindingData.branch) {
+        rowKey += `/${context.bindingData.branch}`;
+    }
+
+    try {
+        const mutationScore = await scoreRepo.selectSingleEntity(`${context.bindingData.provider}/${context.bindingData.owner}`, rowKey);
```

Comment:
The selectSingleEntitty also could return a `null` or `undefined` value right? In that case we should return https://img.shields.io/badge/mutation%20score-unknown-lightgrey.svg i think.

Reply:
I don't think it can - if it can't find the value based on rowkey/partitionkey, it should fail. At least, that's what I expected - will check the DataAccess layer.

Reply:
I would expect it to return `null`. We might need to change the return type to `T | null`. The problem with throwing an error in JavaScript is that it is difficult to distinguish between different kind of errors. I personally also prefer `null`

Reply:
Done